src/mon/MDSMonitor.cc

```diff
@@ -593,10 +593,16 @@ bool MDSMonitor::prepare_beacon(MonOpRequestRef op)
 
   std::set<mds_metric_t> new_types;
```

Comment:
as an aside, we could have `old_types` and `new_types` as `std::unordered_set` instead of `std::set` to optimize on the lookup time